File: Day-01-Bit-Manipulation/05-Min-bit-Filps/solution.cpp
#include <iostream>
using namespace std;

int main() {
    int start, goal, count = 0;
    cin >> start >> goal;

    int num = start ^ goal;

    while (num > 0) {
        if (num & 1)
            count++;
        num = num >> 1;
    }

    cout << count;
    return 0;
}

File: Day-01-Bit-Manipulation/05-Min-bit-Filps/solution.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int start = sc.nextInt();
        int goal = sc.nextInt();
        int count = 0;

        int num = start ^ goal;

        while (num > 0) {
            if ((num & 1) == 1)
                count++;
            num = num >> 1;
        }

        System.out.print(count);
    }
}